src/Microsoft.AspNet.Authentication/SecurityHelper.cs

```diff
@@ -21,16 +19,7 @@ public static class SecurityHelper
         /// <param name="identity"></param>
         public static ClaimsPrincipal MergeUserPrincipal([NotNull] ClaimsPrincipal existingPrincipal, [NotNull] ClaimsPrincipal additionalPrincipal)
         {
-            var newPrincipal = new ClaimsPrincipal();
-            // New principal identities go first
-            newPrincipal.AddIdentities(additionalPrincipal.Identities);
-
-            // Then add any existing non empty or authenticated identities
-            if (existingPrincipal != null)
-            {
-                newPrincipal.AddIdentities(existingPrincipal.Identities.Where(i => i.IsAuthenticated || i.Claims.Count() > 0));
-            }
-            return newPrincipal;
+            return Microsoft.Framework.Internal.SecurityHelper.MergeUserPrincipal(existingPrincipal, additionalPrincipal);
```

Comment:
Pretty much fun stuff like your identity no longer being Authenticated since we do the whole promoting/reordering thing here, and if we do muck with the principal selector it would be in this method as well.